src/python/pants/backend/python/util_rules/lockfile_metadata.py

```diff
@@ -27,21 +27,10 @@ class LockfileMetadata:
     requirements_invalidation_digest: str
     valid_for_interpreter_constraints: InterpreterConstraints
 
-    @staticmethod
-    def new(
-        requirements_invalidation_digest: str,
-        valid_for_interpreter_constraints: InterpreterConstraints,
-    ) -> LockfileMetadata:
-        """Call the most recent version of the `LockfileMetadata` class to construct a concrete
-        instance.
-
-        This static method should be used in place of the `LockfileMetadata` constructor.
-        """
-
-        return LockfileMetadata(requirements_invalidation_digest, valid_for_interpreter_constraints)
-
     @classmethod
-    def from_lockfile(cls, lockfile: bytes) -> LockfileMetadata:
+    def from_lockfile(
+        cls, lockfile: bytes, lockfile_scope_name: str | None = None
```

Comment:
If my read is correct, you always have the `lockfile_path` available. Only for tool lockfiles do you have the `resolve_name` (what you call `lockfile_scope_name`). If you're up for it, the error message would be even better if you could say

> Could not find a Pants metadata block in the lockfile {path} for the resolve `{resolve_name}`.

Or if resolve_name is None,

> Could not find a Pants metadata block in the lockfile {path}.

Reply:
Also please call it `resolve_name`. This error applies to multiple user lockfiles too, and `lockfile_scope_name` is a misnomer there. The way to refer to this is `resolve_name`.

Reply:
Sure thing

Reply:
That makes sense. Making that change.

Reply:
It looks like `lockfile_path` is only available for non-default lockfiles, so I've added `lockfile_path` but made it optional.